docs/proposals/image-provenance.md

```diff
@@ -0,0 +1,358 @@
+    these concepts.  At any rate, Kuberenetes is not ready to be opinionated on these concepts.
+* Sending more information than strictly needed to a third-party service.
+  * Information sent by Kubernetes to a third-party service constitutes an API of Kubernetes, and we want to
+    avoid making these broader than necessary, as it restricts future evolution of Kubernetes, and makes
+    Kubernetes harder to reason about.  Also, excessive information limits cache-ability of decisions.  Caching
+    reduces latency and allows short outages of the backend to be tolerated.
+
+
+Detailed discussion in [Ensuring only images are from approved sources are run](
+https://github.com/kubernetes/kubernetes/issues/22888).
+
+# Implementation
+
+A new admission controller will be added.  That will be the only change.
+
+## Admission controller
+
+An `ImagePolicyWebhook` admission controller will be written.  The admission controller examines all pod objects which are
+created or updated.  It can either admit the pod, or reject it.  If it is rejected, the request sees a `403 FORBIDDEN`
+
+The admission controller code will go in `plugin/pkg/admission/imagepolicy`.
+
+There will be a cache of decisions in the admission controller. 
+
+If the apiserver cannot reach the webhook backend, it will log a warning and either admit or deny the pod.
+A flag will control whether it admits or denys on failure.
+The rationale for deny is that an attacker could DoS the backend or wait for it to be down, and then sneak a
+bad pod into the system.  The rationale for allow here is that, if the cluster admin also does
+after-the-fact auditing of what images were run (which we think will be common), this will catch
+any bad images run during periods of backend failure.  With default-allow, the availability of Kubernetes does
+not depend on the availability of the backend.
+
+# Webhook Backend
+
+The admission controller code in that directory does not contain logic to make an admit/reject decision.  Instead, it extracts
+relevant fields from the Pod creation/update request and sends those fields to a Backend (which we have been loosely calling "WebHooks"
+in Kubernetes).  The request the admission controller sends to the backend is called a WebHook request to distinguish it from the
+request being admission-controlled.  The server that accepts the WebHook request from Kubernetes is called the "Backend"
+to distinguish it from the WebHook request itself, and from the API server.
+
+The whole system will work similarly to the [Authentication WebHook](
+https://github.com/kubernetes/kubernetes/pull/24902
+) or the [AuthorizationWebHook](
+https://github.com/kubernetes/kubernetes/pull/20347).
+
+The WebHook request can optionally authenticate itself to its backend using a token from a `kubeconfig` file.
+
+The WebHook request and response are JSON, and correspond to the following `go` structures:
+
+```go
+// Filename: pkg/apis/authentication.k8s.io/register.go
```

Comment:
Are these types intended to live in the `authentication.k8s.io` API group? Is the package name correct here?